### ISSUE TITLE
Pod volume mounting failing even after PV is bound and attached to pod

### ISSUE THREAD

**Issue by amolsh** [kubernetes/kubernetes#49926]
kubectl version:
Client Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.1", GitCommit:"82450d03cb057bab0950214ef122b67c83fb11df", GitTreeState:"clean", BuildDate:"2016-12-14T00:57:05Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.4+coreos.0", GitCommit:"97c11b097b1a2b194f1eddca8ce5468fcc83331c", GitTreeState:"clean", BuildDate:"2017-03-08T23:54:21Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}

yml file:

```
---
apiVersion: v1
kind: Service
metadata:
  name: nginx
  labels:
    app: nginx
spec:
  ports:
  - port: 80
    name: web
  clusterIP: None
  selector:
    app: nginx
---
apiVersion: apps/v1beta1
kind: StatefulSet
metadata:
  name: web
spec:
  serviceName: "nginx"
  replicas: 2
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - name: nginx
        image: gcr.io/google_containers/nginx-slim:0.8
        ports:
        - containerPort: 80
          name: web
        volumeMounts:
        - name: www
          mountPath: /usr/share/nginx/html
  volumeClaimTemplates:
  - metadata:
      name: www
      annotations:
         volume.beta.kubernetes.io/storage-class: default
    spec:
      accessModes: [ "ReadWriteOnce" ]
      resources:
        requests:
          storage: 1Gi

```
List of bound PVs:

![image](https://user-images.githubusercontent.com/10971917/28811387-77ffde16-76ac-11e7-8cd7-bbc34ff87823.png)

![image](https://user-images.githubusercontent.com/10971917/28811502-41a104c0-76ad-11e7-8c27-06aebd5f7da3.png)

ERROR:
![image](https://user-images.githubusercontent.com/10971917/28811675-6710ccd0-76ae-11e7-89fa-b272024505a1.png)

relevant kubelet logs:

![image](https://user-images.githubusercontent.com/10971917/28811920-cf3e4f34-76af-11e7-9ccc-cc18ee5787aa.png)

**Comment by huangjiasingle** [comment]
@amolsh  you can get the pv?  you storage is use storage-class, if you create pvc ,it's will auto create pv. you can see the pv exist or not

**Comment by amolsh** [comment]
yes i can get it
![image](https://user-images.githubusercontent.com/10971917/28835765-20eb6444-7704-11e7-8af5-914117f38c81.png)

**Comment by huangjiasingle** [comment]
@amolsh did  you created  a storage-class was named default?  cany you show the storage-class?  l think the reason is miss storage-class of named default. do you have any other yaml?

**Comment by amolsh** [comment]
@huangjiasingle  This is my storage class
```
apiVersion: storage.k8s.io/v1beta1
kind: StorageClass
metadata:
  name: default
  annotations:
    storageclass.beta.kubernetes.io/is-default-class: "true"
  labels:
    kubernetes.io/cluster-service: "true"
provisioner: kubernetes.io/aws-ebs
parameters:
  type: gp2

```

![image](https://user-images.githubusercontent.com/10971917/28882620-5d3dd18c-77c9-11e7-82ea-cefd3ca0e9a9.png)

**Comment by jingxu97** [comment]
@amolsh do you have the master log we can take a look? You can email us if your prefer. The kubelet log shows that the volume is not attached to the node yet.

**Comment by amolsh** [comment]
master kube-controll-manager is not generating  any logs related to above statefulset. Also I couldn't find anything in api-server logs also. One more thing I forgot to mention my cluster is on CoreOs machines

**Comment by amolsh** [comment]
@huangjiasingle 
I logs it gives following error, Is is related to IAM policy:

Failed to attach volume "pvc-fff84c66-7b35-11e7-b125-02f3f42ec6aa" on node "ip-100-x-x-x.us-west-2.compute.internal" with: Error attaching EBS volume "vol-00cbb836374d1b37b" to instance "i-03d4ba6b17ab9cf5f": **UnauthorizedOperation: You are not authorized to perform this operation.** Encoded authorization failure message: 4nedpChQKhsxXs......

**Comment by amolsh** [comment]
I updated IAM policy and added  `ec2:AttachVolume` and `ec2:DetachVolume`. It resolved authorization issue. But now it is giving another issue. Even though volume is available attached to instance :

`Failed to attach volume "pvc-79e4c457-7b57-11e7-a96e-0698361089de" on node "ip-100-x-x-x.us-west-2.compute.internal" with: Error attaching EBS volume "vol-00da392489f12395f" to instance "i-02bbbc571c95b69fd": IncorrectState: vol-00da392489f12395f is not 'available'.`

**Comment by amolsh** [comment]
Now volume attachement  is successfull, but still getting error 

![image](https://user-images.githubusercontent.com/10971917/29069431-99ead704-7c58-11e7-98e5-68c4cb0e4824.png)


**master kube-controller logs:**

```

I0808 10:56:41.980229       1 event.go:217] Event(api.ObjectReference{Kind:"StatefulSet", Namespace:"default", Name:"web", UID:"42d0dd3d-7c28-11e7-a194-0acdd643d1f4", APIVersion:"apps", ResourceVersion:"32766795", FieldPath:""}): type: 'Normal' reason: 'SuccessfulCreate' pet: web-0
I0808 10:56:42.010206       1 pet_set.go:332] StatefulSet web blocked from scaling on pod web-0
I0808 10:56:42.028325       1 pet_set.go:332] StatefulSet web blocked from scaling on pod web-0
I0808 10:56:42.045299       1 pet_set.go:332] StatefulSet web blocked from scaling on pod web-0
I0808 10:56:42.074867       1 reconciler.go:213] Started AttachVolume for volume "kubernetes.io/aws-ebs/aws://us-west-2c/vol-041e0445484858feb" to node "ip-100-x-x-216.us-west-2.compute.internal"

I0808 10:56:52.643138       1 operation_executor.go:620] AttachVolume.Attach succeeded for volume "kubernetes.io/aws-ebs/aws://us-west-2c/vol-041e0445484858feb" (spec.Name: "pvc-40b565bc-7c26-11e7-b125-02f3f42ec6aa") from node "ip-100-x-x-216.us-west-2.compute.internal".
I0808 10:56:58.393077       1 pet_set.go:332] StatefulSet web blocked from scaling on pod web-0
I0808 10:57:28.393299       1 pet_set.go:332] StatefulSet web blocked from scaling on pod web-0
I0808 10:57:58.393600       1 pet_set.go:332] StatefulSet web blocked from scaling on pod web-0
I0808 10:58:28.393739       1 pet_set.go:332] StatefulSet web blocked from scaling on pod web-0
I0808 10:58:58.394266       1 pet_set.go:332] StatefulSet web blocked from scaling on pod web-0
I0808 10:59:28.393965       1 pet_set.go:332] StatefulSet web blocked from scaling on pod web-0
I0808 10:59:58.394304       1 pet_set.go:332] StatefulSet web blocked from scaling on pod web-0
```

**Comment by msau42** [comment]
@amolsh are you still seeing this issue?

**Comment by amolsh** [comment]
Yes, it did not resolve, so I stopped trying.

On Nov 9, 2017 10:26 AM, "Michelle Au" <notifications@github.com> wrote:

@amolsh <https://github.com/amolsh> are you still seeing this issue?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/kubernetes/kubernetes/issues/49926#issuecomment-343047278>,
or mute the thread

**Comment by msau42** [comment]
This seems like a possible issue related to EBS volumes.  Can @kubernetes/sig-aws-misc help out?

**Comment by mattcamp** [comment]
I'm also having this issue: "**AttachVolume.Attach failed for volume "pvc-5aa1db99-d04b-11e7-96e1-0a328f684a08" : Error attaching EBS volume "vol-026ef055a9d715c14" to instance "i-0e6f2c5d8edd415e4": IncorrectState: vol-026ef055a9d715c14 is not 'available'. status code: 400**"

Using a single master, 3 node cluster in AWS (deployed via KOPS)... trying to deploy influxdb via helm.

It worked the first time I deployed it but after deleting the deployment (to change collectd config) I can't get it to work... it doesn't seem to matter which node, none of them work.

Grafana is deployed on the same cluster just fine (but have only deployed it once).

`helm install stable/influxdb --name influxdb --set persistence.enabled=true --set config.collectd.enabled=true --namespace poke`

`Client Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.2", GitCommit:"922a86cfcd65915a9b2f69f3f193b8907d741d9c", GitTreeState:"clean", BuildDate:"2017-07-21T08:23:22Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.10", GitCommit:"bebdeb749f1fa3da9e1312c4b08e439c404b3136", GitTreeState:"clean", BuildDate:"2017-11-03T16:31:49Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}`

[kubectl.log](https://github.com/kubernetes/kubernetes/files/1499081/kubectl.log)
[node.log](https://github.com/kubernetes/kubernetes/files/1499082/node.log)

**Comment by gnufied** [comment]
The `IncorrectState: vol-026ef055a9d715c14 is not 'available'. status code: 400"` is a transient error that happens when you try to attach a volume immediately after creating it, it goes away in half a second or so and controller will keep retrying attach. 

@mattcamp are you seeing "Not available" error more than once in controller logs? Looking at kubectl output I see following message:

```
  3m            3m      1       kubelet, ip-172-20-36-162.eu-west-1.compute.internal                                            Normal  SuccessfulMountVolume   MountVolume.SetUp succeeded for volume "pvc-5aa1db99-d04b-11e7-96e1-0a328f684a08"
``

Which indicates attach succeeded and volume was attached to the node. I think something else happened when container was started:

```
  3m            4s      21      kubelet, ip-172-20-36-162.eu-west-1.compute.internal    spec.containers{influxdb-influxdb}      Warning BackOff                 Back-off restarting failed container
```

You may want to look into it.

**Comment by neverfox** [comment]
Ran into this when trying a stateful set in 1.8 that works great in 1.7. It does *not* retry and instead puts the pod into a CreateContainerConfigError state (something that's new to me).

**Comment by feffi** [comment]
Hi, i got the same (?) error when trying 'helm install'

kubectl get storageclass
NAME            PROVISIONER             AGE
default         kubernetes.io/aws-ebs   4d
gp2 (default)   kubernetes.io/aws-ebs   4d

kubectl get persistentvolume
NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS    CLAIM                          STORAGECLASS   REASON    AGE
pvc-8b23869c-f196-11e7-b93f-020c644f020c   8Gi        RWO            Delete           Bound     default/sonarqube-postgresql   gp2                      3d
pvc-c4707186-f3f0-11e7-b93f-020c644f020c   20Gi       RWO            Delete           Bound     default/git-minio              gp2                      21m
pvc-c4710474-f3f0-11e7-b93f-020c644f020c   10Gi       RWO            Delete           Bound     default/git-postgresql         gp2                      21m
pvc-c471bf13-f3f0-11e7-b93f-020c644f020c   10Gi       RWO            Delete           Bound     default/git-redis              gp2                      21m
pvc-c47273d1-f3f0-11e7-b93f-020c644f020c   10Gi       RWO            Delete           Bound     default/git-registry-data      gp2                      21m
pvc-c4732d10-f3f0-11e7-b93f-020c644f020c   10Gi       RWO            Delete           Bound     default/git-gitlab-data        gp2                      21m

kubectl get persistentvolumeclaim
NAME                   STATUS    VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
git-gitlab-data        Bound     pvc-c4732d10-f3f0-11e7-b93f-020c644f020c   10Gi       RWO            gp2            21m
git-minio              Bound     pvc-c4707186-f3f0-11e7-b93f-020c644f020c   20Gi       RWO            gp2            21m
git-postgresql         Bound     pvc-c4710474-f3f0-11e7-b93f-020c644f020c   10Gi       RWO            gp2            21m
git-redis              Bound     pvc-c471bf13-f3f0-11e7-b93f-020c644f020c   10Gi       RWO            gp2            21m
git-registry-data      Bound     pvc-c47273d1-f3f0-11e7-b93f-020c644f020c   10Gi       RWO            gp2            21m
sonarqube-postgresql   Bound     pvc-8b23869c-f196-11e7-b93f-020c644f020c   8Gi        RWO            gp2            3d

Events:
  Type     Reason                 Age                 From                                                    Message
  ----     ------                 ----                ----                                                    -------
  Warning  FailedScheduling       24m (x2 over 24m)   default-scheduler                                       PersistentVolumeClaim is not bound: "git-registry-data" (repeated 3 times)
  Normal   Scheduled              24m                 default-scheduler                                       Successfully assigned git-registry-7b784bb55d-lvhbk to ip-172-20-34-53.eu-central-1.compute.internal
  Warning  FailedMount            24m                 attachdetach                                            AttachVolume.Attach failed for volume "pvc-c47273d1-f3f0-11e7-b93f-020c644f020c" : Error attaching EBS volume "vol-0e7150ac4806512a8" to instance "i-0ee36c5c18824a89e": "IncorrectState: vol-0e7150ac4806512a8 is not 'available'.\n\tstatus code: 400, request id: c8cc4773-af8a-402c-b08b-fb280b4fb3da"
  Warning  FailedMount            23m                 attachdetach                                            AttachVolume.Attach failed for volume "pvc-c47273d1-f3f0-11e7-b93f-020c644f020c" : Error attaching EBS volume "vol-0e7150ac4806512a8" to instance "i-0ee36c5c18824a89e": "IncorrectState: vol-0e7150ac4806512a8 is not 'available'.\n\tstatus code: 400, request id: 1e0a0209-147c-458f-9fbf-7c45ef885e7c"
  Normal   SuccessfulMountVolume  23m                 kubelet, ip-172-20-34-53.eu-central-1.compute.internal  MountVolume.SetUp succeeded for volume "default-token-z2zbm"
  Normal   SuccessfulMountVolume  23m                 kubelet, ip-172-20-34-53.eu-central-1.compute.internal  MountVolume.SetUp succeeded for volume "pvc-c47273d1-f3f0-11e7-b93f-020c644f020c"
  Warning  FailedMount            12m (x5 over 21m)   kubelet, ip-172-20-34-53.eu-central-1.compute.internal  Unable to mount volumes for pod "git-registry-7b784bb55d-lvhbk_default(c4a17555-f3f0-11e7-b93f-020c644f020c)": timeout expired waiting for volumes to attach/mount for pod "default"/"git-registry-7b784bb55d-lvhbk". list of unattached/unmounted volumes=[certs]
  Warning  FailedMount            11m (x14 over 23m)  kubelet, ip-172-20-34-53.eu-central-1.compute.internal  MountVolume.SetUp failed for volume "certs" : secrets "registry-server-tls" not found
  Warning  FailedSync             3m (x9 over 21m)    kubelet, ip-172-20-34-53.eu-central-1.compute.internal  Error syncing pod

**Comment by msau42** [comment]
@feffi the mount error you got is pretty self-explanatory:
```
MountVolume.SetUp failed for volume "certs" : secrets "registry-server-tls" not found
```

This is different from the OP's error, which is related to EBS volumes.

**Comment by macropin** [comment]
@msau42 I think he's referring to
 `  Warning  FailedScheduling       24m (x2 over 24m)   default-scheduler                                       PersistentVolumeClaim is not bound: "git-registry-data" (repeated 3 times)`
We're also seeing this. Just started happening after upgrading CoreOS and moving to `m4.large` instance types (EBS optimised).

**Comment by macropin** [comment]
It's worth noting that a message such`  Warning  FailedScheduling       24m (x2 over 24m)   default-scheduler                                       PersistentVolumeClaim is not bound: "git-registry-data" (repeated 3 times)` can appear to indicate a PV/PVC issue, when infact the pod is restarting due to error and the message is repeated when the container is rescheduled.

**Comment by hugohenley** [comment]
Same issue here!

**Comment by jolcese** [comment]
Same problem when deploying a Stateful Set on kops on AWS.
I;m trying to deploy Cassandra with 6 replicas and sometimes it fails on the first pod and sometimes in the 2nd...

```
PersistentVolumeClaim is not bound: "cassandra-data-pvc-cassandra-1" (repeated 9 times)
AttachVolume.Attach failed for volume "pvc-dd201a20-06dd-11e8-9f82-06210fc81b88" : Error attaching EBS volume "vol-0a752df5ddef30f64" to instance "i-039b684aca16fc74e": "IncorrectState: vol-0a752df5ddef30f64 is not 'available'.\n\tstatus code: 400, request id: e57f9b75-e57b-4455-843c-9f9b61eb83f1"
Back-off restarting failed container
Error syncing pod
```

**Comment by lvicentesanchez** [comment]
Same error here... I'm create an storage class, pv, pvc and a statefulset in one go and I get the same error. The ebs volume it's actually attached to the node so I'm not sure what's going on.

**Comment by lvicentesanchez** [comment]
I have tried to remove the pod and it still fails, once recreated.

**Comment by lvicentesanchez** [comment]
I'm using Kubernetes 1.8.8, deployed using kops 1.8.1, with RBAC enabled.

**Comment by lvicentesanchez** [comment]
Some additional information. If I remove everything using `kubectl delete -f mantifest/influxdb.yaml`, the EBS volume is in available status. If then I create it again, I get a timeout while trying to mount the volume but the EBS volume is 'in use'. So... first time, I get an error because of the 'available' status, after that... the volume can't be attached even  if it's 'in use' by the target node.

**Comment by lvicentesanchez** [comment]
For what is worth... here is the manifest I'm using:

```yaml
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: monitoring-influxdb
provisioner: kubernetes.io/aws-ebs
parameters:
  type: gp2
---
apiVersion: apps/v1beta1
kind: StatefulSet
metadata:
  name: monitoring-influxdb
  namespace: kube-system
  labels:
    k8s-app: monitoring-influxdb
    version: v1
    kubernetes.io/cluster-service: "true"
    addonmanager.kubernetes.io/mode: Reconcile
spec:
  serviceName: "monitoring-influxdb"
  replicas: 1
  template:
    metadata:
      labels:
        task: monitoring
        k8s-app: monitoring-influxdb
        version: v1
        kubernetes.io/cluster-service: "true"
    spec:
      tolerations:
      - key: "datalake/dedicated"
        operator: "Equal"
        value: "monitoring-influxdb"
        effect: "NoSchedule"
      containers:
      - name: influxdb
        image: k8s.gcr.io/heapster-influxdb-amd64:v1.3.3
        volumeMounts:
        - mountPath: /data
          name: influxdb-persistent-storage
      nodeSelector:
        kops.k8s.io/instancegroup: influxdb
  volumeClaimTemplates:
  - metadata:
      name: influxdb-persistent-storage
      annotations:
        volume.beta.kubernetes.io/storage-class: monitoring-influxdb
    spec:
      accessModes: [ "ReadWriteOnce" ]
      resources:
        requests:
          storage: 500Gi
---
apiVersion: v1
kind: Service
metadata:
  labels:
    task: monitoring
    kubernetes.io/cluster-service: "true"
    kubernetes.io/name: monitoring-influxdb
  name: monitoring-influxdb
  namespace: kube-system
spec:
  ports:
  - port: 8086
    targetPort: 8086
  selector:
    k8s-app: monitoring-influxdb
```

**Comment by lvicentesanchez** [comment]
This is the log I get when I first create the stateful set:

```
I0220 11:35:11.373594       6 event.go:218] Event(v1.ObjectReference{Kind:"StatefulSet", Namespace:"kube-system", Name:"monitoring-influxdb", UID:"1c4aa40e-1632-11e8-8142-0a3a1b5232ac", APIVersion:"apps", ResourceVersion:"462706", FieldPath:""}): type: 'Normal' reason: 'SuccessfulCreate' create Claim influxdb-persistent-storage-monitoring-influxdb-0 Pod monitoring-influxdb-0 in StatefulSet monitoring-influxdb success
I0220 11:35:11.379432       6 event.go:218] Event(v1.ObjectReference{Kind:"StatefulSet", Namespace:"kube-system", Name:"monitoring-influxdb", UID:"1c4aa40e-1632-11e8-8142-0a3a1b5232ac", APIVersion:"apps", ResourceVersion:"462706", FieldPath:""}): type: 'Normal' reason: 'SuccessfulCreate' create Pod monitoring-influxdb-0 in StatefulSet monitoring-influxdb successful
I0220 11:35:11.462944       6 service_controller.go:323] Not persisting unchanged LoadBalancerStatus for service kube-system/monitoring-influxdb to registry.
I0220 11:35:11.537592       6 aws.go:1246] Found instances in zones map[eu-west-1c:{} eu-west-1a:{} eu-west-1b:{}]
I0220 11:35:11.537621       6 util.go:385] Detected StatefulSet-style volume name "influxdb-persistent-storage-monitoring-influxdb-0"; index=0
I0220 11:35:11.537632       6 util.go:314] Creating volume for PVC "influxdb-persistent-storage-monitoring-influxdb-0"; chose zone="eu-west-1a" from zones=["eu-west-1a" "eu-west-1b" "eu-west-1c"]
I0220 11:35:11.828239       6 aws_util.go:140] Successfully created EBS Disk volume aws://eu-west-1a/vol-0d204f81da2705f51
I0220 11:35:11.884363       6 pv_controller.go:1400] volume "pvc-1c4c0456-1632-11e8-8142-0a3a1b5232ac" provisioned for claim "kube-system/influxdb-persistent-storage-monitoring-influxdb-0"
I0220 11:35:11.893032       6 pv_controller.go:710] volume "pvc-1c4c0456-1632-11e8-8142-0a3a1b5232ac" entered phase "Bound"
I0220 11:35:11.893341       6 pv_controller.go:831] volume "pvc-1c4c0456-1632-11e8-8142-0a3a1b5232ac" bound to claim "kube-system/influxdb-persistent-storage-monitoring-influxdb-0"
I0220 11:35:11.902229       6 pv_controller.go:654] claim "kube-system/influxdb-persistent-storage-monitoring-influxdb-0" entered phase "Bound"
I0220 11:35:12.437339       6 reconciler.go:277] attacherDetacher.AttachVolume started for volume "pvc-1c4c0456-1632-11e8-8142-0a3a1b5232ac" (UniqueName: "kubernetes.io/aws-ebs/aws://eu-west-1a/vol-0d204f81da2705f51") from node "ip-10-4-34-153.eu-west-1.compute.internal" 
I0220 11:35:12.479038       6 aws.go:1435] Assigned mount device bi -> volume vol-0d204f81da2705f51
I0220 11:35:12.752942       6 aws.go:1458] Releasing in-process attachment entry: bi -> volume vol-0d204f81da2705f51
E0220 11:35:12.752964       6 attacher.go:74] Error attaching volume "aws://eu-west-1a/vol-0d204f81da2705f51" to node "ip-10-4-34-153.eu-west-1.compute.internal": Error attaching EBS volume "vol-0d204f81da2705f51" to instance "i-0cbeea0215856d50f": "IncorrectState: vol-0d204f81da2705f51 is not 'available'.\n\tstatus code: 400, request id: 4181e4eb-5458-4f85-8fc4-f37feb99a9e5"
E0220 11:35:12.753116       6 nestedpendingoperations.go:264] Operation for "\"kubernetes.io/aws-ebs/aws://eu-west-1a/vol-0d204f81da2705f51\"" failed. No retries permitted until 2018-02-20 11:35:13.253099099 +0000 UTC (durationBeforeRetry 500ms). Error: AttachVolume.Attach failed for volume "pvc-1c4c0456-1632-11e8-8142-0a3a1b5232ac" (UniqueName: "kubernetes.io/aws-ebs/aws://eu-west-1a/vol-0d204f81da2705f51") from node "ip-10-4-34-153.eu-west-1.compute.internal" : Error attaching EBS volume "vol-0d204f81da2705f51" to instance "i-0cbeea0215856d50f": "IncorrectState: vol-0d204f81da2705f51 is not 'available'.\n\tstatus code: 400, request id: 4181e4eb-5458-4f85-8fc4-f37feb99a9e5"
I0220 11:35:12.753307       6 event.go:218] Event(v1.ObjectReference{Kind:"Pod", Namespace:"kube-system", Name:"monitoring-influxdb-0", UID:"1c4c4f8f-1632-11e8-8142-0a3a1b5232ac", APIVersion:"v1", ResourceVersion:"462720", FieldPath:""}): type: 'Warning' reason: 'FailedMount' AttachVolume.Attach failed for volume "pvc-1c4c0456-1632-11e8-8142-0a3a1b5232ac" : Error attaching EBS volume "vol-0d204f81da2705f51" to instance "i-0cbeea0215856d50f": "IncorrectState: vol-0d204f81da2705f51 is not 'available'.\n\tstatus code: 400, request id: 4181e4eb-5458-4f85-8fc4-f37feb99a9e5"
I0220 11:35:13.339127       6 reconciler.go:277] attacherDetacher.AttachVolume started for volume "pvc-1c4c0456-1632-11e8-8142-0a3a1b5232ac" (UniqueName: "kubernetes.io/aws-ebs/aws://eu-west-1a/vol-0d204f81da2705f51") from node "ip-10-4-34-153.eu-west-1.compute.internal" 
I0220 11:35:13.420776       6 aws.go:1435] Assigned mount device ct -> volume vol-0d204f81da2705f51
I0220 11:35:13.617087       6 aws.go:1458] Releasing in-process attachment entry: ct -> volume vol-0d204f81da2705f51
E0220 11:35:13.617107       6 attacher.go:74] Error attaching volume "aws://eu-west-1a/vol-0d204f81da2705f51" to node "ip-10-4-34-153.eu-west-1.compute.internal": Error attaching EBS volume "vol-0d204f81da2705f51" to instance "i-0cbeea0215856d50f": "IncorrectState: vol-0d204f81da2705f51 is not 'available'.\n\tstatus code: 400, request id: af8a420e-c3e4-4781-8a5f-718bccfd5c3c"
E0220 11:35:13.617223       6 nestedpendingoperations.go:264] Operation for "\"kubernetes.io/aws-ebs/aws://eu-west-1a/vol-0d204f81da2705f51\"" failed. No retries permitted until 2018-02-20 11:35:14.617205978 +0000 UTC (durationBeforeRetry 1s). Error: AttachVolume.Attach failed for volume "pvc-1c4c0456-1632-11e8-8142-0a3a1b5232ac" (UniqueName: "kubernetes.io/aws-ebs/aws://eu-west-1a/vol-0d204f81da2705f51") from node "ip-10-4-34-153.eu-west-1.compute.internal" : Error attaching EBS volume "vol-0d204f81da2705f51" to instance "i-0cbeea0215856d50f": "IncorrectState: vol-0d204f81da2705f51 is not 'available'.\n\tstatus code: 400, request id: af8a420e-c3e4-4781-8a5f-718bccfd5c3c"
I0220 11:35:13.617390       6 event.go:218] Event(v1.ObjectReference{Kind:"Pod", Namespace:"kube-system", Name:"monitoring-influxdb-0", UID:"1c4c4f8f-1632-11e8-8142-0a3a1b5232ac", APIVersion:"v1", ResourceVersion:"462720", FieldPath:""}): type: 'Warning' reason: 'FailedMount' AttachVolume.Attach failed for volume "pvc-1c4c0456-1632-11e8-8142-0a3a1b5232ac" : Error attaching EBS volume "vol-0d204f81da2705f51" to instance "i-0cbeea0215856d50f": "IncorrectState: vol-0d204f81da2705f51 is not 'available'.\n\tstatus code: 400, request id: af8a420e-c3e4-4781-8a5f-718bccfd5c3c"
I0220 11:35:14.641633       6 reconciler.go:277] attacherDetacher.AttachVolume started for volume "pvc-1c4c0456-1632-11e8-8142-0a3a1b5232ac" (UniqueName: "kubernetes.io/aws-ebs/aws://eu-west-1a/vol-0d204f81da2705f51") from node "ip-10-4-34-153.eu-west-1.compute.internal" 
I0220 11:35:14.752566       6 aws.go:1435] Assigned mount device cb -> volume vol-0d204f81da2705f51
I0220 11:35:14.963745       6 aws.go:1708] AttachVolume volume="vol-0d204f81da2705f51" instance="i-0cbeea0215856d50f" request returned {
  AttachTime: 2018-02-20 11:35:14.94 +0000 UTC,
  Device: "/dev/xvdcb",
  InstanceId: "i-0cbeea0215856d50f",
  State: "attaching",
  VolumeId: "vol-0d204f81da2705f51"
}
I0220 11:35:15.044813       6 aws.go:1557] Waiting for volume "vol-0d204f81da2705f51" state: actual=attaching, desired=attached
I0220 11:35:25.143085       6 aws.go:1458] Releasing in-process attachment entry: cb -> volume vol-0d204f81da2705f51
I0220 11:35:25.143118       6 operation_generator.go:278] AttachVolume.Attach succeeded for volume "pvc-1c4c0456-1632-11e8-8142-0a3a1b5232ac" (UniqueName: "kubernetes.io/aws-ebs/aws://eu-west-1a/vol-0d204f81da2705f51") from node "ip-10-4-34-153.eu-west-1.compute.internal" 
```

**Comment by lvicentesanchez** [comment]
And these are the events I get from the pod:

```
Events:
  Type     Reason                 Age              From                                                Message
  ----     ------                 ----             ----                                                -------
  Warning  FailedScheduling       3m (x2 over 3m)  default-scheduler                                   PersistentVolumeClaim is not bound: "influxdb-persistent-storage-monitoring-influxdb-0" (repeated 5 times)
  Normal   Scheduled              3m               default-scheduler                                   Successfully assigned monitoring-influxdb-0 to ip-10-4-34-153.eu-west-1.compute.internal
  Normal   SuccessfulMountVolume  3m               kubelet, ip-10-4-34-153.eu-west-1.compute.internal  MountVolume.SetUp succeeded for volume "default-token-khwqh"
  Warning  FailedMount            3m               attachdetach                                        AttachVolume.Attach failed for volume "pvc-1c4c0456-1632-11e8-8142-0a3a1b5232ac" : Error attaching EBS volume "vol-0d204f81da2705f51" to instance "i-0cbeea0215856d50f": "IncorrectState: vol-0d204f81da2705f51 is not 'available'.\n\tstatus code: 400, request id: 4181e4eb-5458-4f85-8fc4-f37feb99a9e5"
  Warning  FailedMount            3m               attachdetach                                        AttachVolume.Attach failed for volume "pvc-1c4c0456-1632-11e8-8142-0a3a1b5232ac" : Error attaching EBS volume "vol-0d204f81da2705f51" to instance "i-0cbeea0215856d50f": "IncorrectState: vol-0d204f81da2705f51 is not 'available'.\n\tstatus code: 400, request id: af8a420e-c3e4-4781-8a5f-718bccfd5c3c"
  Warning  FailedMount            1m               kubelet, ip-10-4-34-153.eu-west-1.compute.internal  Unable to mount volumes for pod "monitoring-influxdb-0_kube-system(1c4c4f8f-1632-11e8-8142-0a3a1b5232ac)": timeout expired waiting for volumes to attach/mount for pod "kube-system"/"monitoring-influxdb-0". list of unattached/unmounted volumes=[influxdb-persistent-storage]
  Warning  FailedSync             1m               kubelet, ip-10-4-34-153.eu-west-1.compute.internal  Error syncing pod
```

**Comment by lvicentesanchez** [comment]
And these are the events when I delete the pod and let the statefulset to recreate it:

```
Events:
  Type     Reason                 Age               From                                                Message
  ----     ------                 ----              ----                                                -------
  Normal   Scheduled              4m                default-scheduler                                   Successfully assigned monitoring-influxdb-0 to ip-10-4-34-153.eu-west-1.compute.internal
  Normal   SuccessfulMountVolume  4m                kubelet, ip-10-4-34-153.eu-west-1.compute.internal  MountVolume.SetUp succeeded for volume "default-token-khwqh"
  Warning  FailedMount            40s (x2 over 2m)  kubelet, ip-10-4-34-153.eu-west-1.compute.internal  Unable to mount volumes for pod "monitoring-influxdb-0_kube-system(27bcb2f9-1633-11e8-8142-0a3a1b5232ac)": timeout expired waiting for volumes to attach/mount for pod "kube-system"/"monitoring-influxdb-0". list of unattached/unmounted volumes=[influxdb-persistent-storage]
  Warning  FailedSync             40s (x2 over 2m)  kubelet, ip-10-4-34-153.eu-west-1.compute.internal  Error syncing pod
```

**Comment by snoby** [comment]
My understanding is that that there are some real issues with mounting volumes on aws based k8s clusters that v1.9.0 and 1.10 are suppose to address.  Mainly because Amazon is making K8s a first class citizen with EKS.  

I had the same issues and moved my storage to EFS until the EBS issues are resolved.  I have not had issues with EFS.  My $0.02

**Comment by lvicentesanchez** [comment]
@snoby using this https://github.com/kubernetes-incubator/external-storage/tree/master/aws/efs?

**Comment by snoby** [comment]
Yes.

**Comment by lvicentesanchez** [comment]
For the record... I just created a 1.7.12 cluster using kops 1.8.1 and same manifests just work so... something is seriously broken in the 1.8.x release.

**Comment by msau42** [comment]
@lvicentesanchez did you bring up your cluster with alpha feature gates?  There was a known issue with alpha clusters using StatefulSets and zonal PVCs in 1.8, and has been fixed in a later patch release.  #56265

**Comment by lvicentesanchez** [comment]
Ok... I found out why I was having problems... the instance type of the node where the PV has to be attached was m5.4xlarge and, according to kops release notes, attaching PV to NVME instances is only supported by kubernetes 1.9.x so... anyone having this problem, make sure you are not facing having the same problem.

**Comment by Tarun17github** [comment]
@lvicentesanchez Thanks a ton, this is exactly what turned out for us. We spawned new cluster with C5.9xlarge ( current generation ) instances on AWS which was getting volumes with NVME conventions which kuberntes 1.8 failed to understand on Rancher. Creating new cluster with older instance type with xvd** convention has solved the issue for us.

**Comment by shredder-rull** [comment]
Have the same issue for c5 and m5 instances. Works on c4
`{
  "clientVersion": {
    "major": "1",
    "minor": "10",
    "gitVersion": "v1.10.1",
    "gitCommit": "d4ab47518836c750f9949b9e0d387f20fb92260b",
    "gitTreeState": "clean",
    "buildDate": "2018-04-13T22:29:03Z",
    "goVersion": "go1.9.5",
    "compiler": "gc",
    "platform": "darwin/amd64"
  },
  "serverVersion": {
    "major": "1",
    "minor": "9",
    "gitVersion": "v1.9.3",
    "gitCommit": "d2835416544f298c919e2ead3be3d0864b52323b",
    "gitTreeState": "clean",
    "buildDate": "2018-02-07T11:55:20Z",
    "goVersion": "go1.9.2",
    "compiler": "gc",
    "platform": "linux/amd64"
  }
}`
Kops version: 1.9.0

**Comment by cyrillk** [comment]
We experience the same with `m5.large` on `eu-west-3`.

**Comment by senorequeso** [comment]
Similar story.`m5.large` and `m5.xlarge` did not work on `us-east-2a`. `t2.large` worked perfectly though and my stateful set was able to be made. 

Kubernetes version: `1.8.7`

**Comment by msau42** [comment]
@gnufied @jsafrane seems like an aws ebs issue

**Comment by msau42** [comment]
@kubernetes/sig-storage-bugs

**Comment by u2mejc** [comment]
Of note, I'm no longer able to reproduce this issue in 1.9.8, but there is a 45sec back off where the pod claims `PersistentVolumeClaim is not bound: "test-StatefulSet" (repeated 10 times)` when it first is created, event thought you can clearly see the PVC was instantly successful.

Tested: us-west-2, m5.large, K8 v1.9.8

**Comment by redbaron** [comment]
@u2mejc it looks suspiciously similar to what we see on bare-metal,  it looks like attach-detach controller can take minutes to update node status. See details in https://github.com/kubernetes/kubernetes/issues/64549  looks like it see new value only after some internal timeout, it then relists resources and sees update.

**Comment by u2mejc** [comment]
thank you for touching this issue @redbaron, and I have to admit that since I have posted the above, I've found the behavior to be extremely unreliable and hard to reproduce a successful binding. 😨 My success rate of getting this to work is all over.

**Comment by greenkiwi** [comment]
We're having the same problems.  It was certainly a non-obvious problem to track down.  I found it by seeing the description of the problem in a stackoverflow issue:
https://stackoverflow.com/questions/49661306/timeout-mounting-pvc-volume-to-pod

Is there anything else that we can do to help tack down the issue?

**Comment by jiashenC** [comment]
I am facing the same issue for instance type `m4.xlarge`. Anyone has the problem for `m4.xlarge` machine?

**Comment by bobo** [comment]
had same problem, using kops 1.9.1 and kubernetes 1.9.6, changing from m5.medium to t2.large instances worked for me.

**Comment by Rouche** [comment]
Sorry for my n00b question in advance. I am just starting using kops and got this issue too on us-east-2a.

How do you specify t2.large? This is my create command:
```kops create cluster kitfox-k8s-test.k8s.local --zones us-east-2a --yes```

Thanks.

**Comment by devnulled** [comment]
@Rouche `--master-size` and `--node-size`

**Comment by coryodaniel** [comment]
Having the same issue. Change from `m5` to `t2`s fixed it. Would love a fix for m5s. <3

**Comment by u2mejc** [comment]
Had to downgrade from M5 to M4, because shared T2 instances are not acceptable in our production.

**Comment by turgayozgur** [comment]
Here is how I fixed.

Kubernetes version: 1.9.10
Machine type: c5.xlarge

[kops 1.8 release notes](https://github.com/kubernetes/kops/blob/master/docs/releases/1.8-NOTES.md#significant-changes)

> New AWS instance types: P3, C5, M5, H1. Please note that NVME volumes are not supported on the default jessie image, so masters will not boot on M5 and C5 instance types unless a stretch image is chosen (change stretch to jessie in the image name). Also note that kubernetes will not support mounting persistent volumes on NVME instances until Kubernetes v1.9.

`kops edit ig nodes`

> Just change the image from jessie to stretch:

from: 
`image: kope.io/k8s-1.9-debian-jessie-amd64-hvm-ebs-2018-03-11`
to: 
`image: kope.io/k8s-1.9-debian-stretch-amd64-hvm-ebs-2018-03-11`

> Note: Make sure you have installed kubernetes version 1.9.x or upper.

**Comment by iamShantanu101** [comment]
> Here is how I fixed.
> 
> Kubernetes version: 1.9.10
> Machine type: c5.xlarge
> 
> [kops 1.8 release notes](https://github.com/kubernetes/kops/blob/master/docs/releases/1.8-NOTES.md#significant-changes)
> 
> > New AWS instance types: P3, C5, M5, H1. Please note that NVME volumes are not supported on the default jessie image, so masters will not boot on M5 and C5 instance types unless a stretch image is chosen (change stretch to jessie in the image name). Also note that kubernetes will not support mounting persistent volumes on NVME instances until Kubernetes v1.9.
> 
> `kops edit ig nodes`
> 
> > Just change the image from jessie to stretch:
> 
> from:
> `image: kope.io/k8s-1.9-debian-jessie-amd64-hvm-ebs-2018-03-11`
> to:
> `image: kope.io/k8s-1.9-debian-stretch-amd64-hvm-ebs-2018-03-11`
> 
> > Note: Make sure you have installed kubernetes version 1.9.x or upper.

@turgayozgur This indeed fixed the issue. Thanks. Saved me a lot of time.

**Comment by omerh** [comment]
I am also having issues with nvme instances, in this case, I tried using m5.large with pvc.
This is the StatefulSet that reproduce this

```
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: k8s-rmq
spec:
  serviceName: "k8s-rmq"
  replicas: 1
  selector:
    matchLabels:
      app: k8s-rmq
  template:
    metadata:
      labels:
        app: k8s-rmq
      annotations:
        cluster-autoscaler.kubernetes.io/safe-to-evict: "false"
    spec:
      nodeSelector:
        kops.k8s.io/instancegroup: nodes
      terminationGracePeriodSeconds: 30
      containers:
      - name: k8s-rmq
        imagePullPolicy: IfNotPresent
        image: rabbitmq:3.7.8-management-alpine
        ports:
        - containerPort: 5672
          name: amqp
        - containerPort: 15672
          name: management
        envFrom:
            - configMapRef:
                name: k8s-dev-aws         
        env:
          - name: RABBITMQ_DEFAULT_USER
            value: example
          - name: RABBITMQ_DEFAULT_PASS
            value: example
        resources:
          limits:
            cpu: "800m"
            memory: "1Gi"
          requests:
            cpu: "100m"
            memory: "128Mi"
        livenessProbe:
          tcpSocket:
            port: 5672
          initialDelaySeconds: 20
          timeoutSeconds: 5
          periodSeconds: 30
          failureThreshold: 2
          successThreshold: 1
        readinessProbe:
          tcpSocket:
            port: 5672
          initialDelaySeconds: 20
          timeoutSeconds: 5
          periodSeconds: 30
          failureThreshold: 2
          successThreshold: 1
        volumeMounts:
        - name: rmqvol
          mountPath: /var/lib/rabbitmq
  volumeClaimTemplates:
  - metadata:
      name: rmqvol
    spec:
      accessModes: [ "ReadWriteOnce" ]
      resources:
        requests:
          storage: 20Gi
```

this is the storage classes:

```
NAME            PROVISIONER             AGE
default         kubernetes.io/aws-ebs   337d
gp2 (default)   kubernetes.io/aws-ebs   337d
```

The EBS is created and attached to the instance.

but Kubelet fails to mount the disk into pod

```
1m          1m           1       k8s-rmq-0.1573f1a3938f660f                                     Pod                                                             Warning   FailedMount                    kubelet, ip-172-20-57-150.eu-west-1.compute.internal      Unable to mount volumes for pod "k8s-rmq-0_default(1c45f9a0-0932-11e9-b1e7-0ac8a16a5f0c)": timeout expired waiting for volumes to attach or mount for pod "default"/"k8s-rmq-0". list of unmounted volumes=[rmqvol default-token-wp48g]. list of unattached volumes=[rmqvol default-token-wp48g]
```

Cluster provisioned using Kops 1.10

kubectl version

```
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.0", GitCommit:"ddf47ac13c1a9483ea035a79cd7c10005ff21a6d", GitTreeState:"clean", BuildDate:"2018-12-03T21:04:45Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.11", GitCommit:"637c7e288581ee40ab4ca210618a89a555b6e7e9", GitTreeState:"clean", BuildDate:"2018-11-26T14:25:46Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

When checking on m5.large the mounts on the node there is not disk mount on the nvme drive.

when replacing to m4.large the mounts has:
`/dev/xvdcu       20G   49M   19G   1% /var/lib/kubelet/plugins/kubernetes.io/aws-ebs/mounts/aws/eu-west-1a/vol-0b73b3a1bf15aac39`

The node image in Kops is: `kope.io/k8s-1.10-debian-jessie-amd64-hvm-ebs-2018-08-17`

And on the same note, different use case, when launching a new cluster using Kops and masters are nvme instances like m5.large, the host startup fails to mount the etcd volume and hangs with protokube:1.10.0 in a loop for 

```
I1226 20:23:59.194888    1721 aws_volume.go:320] nvme path not found "/rootfs/dev/disk/by-id/nvme-Amazon_Elastic_Block_Store_vol0fdeaa59d34bd2ab1"
```

After installing name-cli I can see that the volume exists 

```
root@ip-10-101-35-149:~# nvme list
Node             SN                   Model                                    Namespace Usage                      Format           FW Rev
---------------- -------------------- ---------------------------------------- --------- -------------------------- ---------------- --------
/dev/nvme0n1     vol0913974dacc67c490 Amazon Elastic Block Store               1           0.00   B /  68.72  GB    512   B +  0 B   1.0
/dev/nvme1n1     vol0fdeaa59d34bd2ab1 Amazon Elastic Block Store               1           0.00   B /  21.47  GB    512   B +  0 B   1.0
/dev/nvme2n1     vol0a587f2950331bf7b Amazon Elastic Block Store               1           0.00   B /  21.47  GB    512   B +  0 B   1.0
```

But this /dev/disk/by-id/nvme-Amazon_Elastic_Block_Store_vol0fdeaa59d34bd2ab1 not exists.

the disk mapping /dev/disk/by-id doesn't exists only /dev/disk/by-uuid/

So basically I can use any nvme base instance for masters or nodes that has a EBS pvc

**Comment by chey** [comment]
Cloud: AWS
OS: RedHat 7.6
kube version: v1.12.5

Having a similar problem with storage myself. When draining a node that has an EBS volume which is attached to a Pod and/or Deployment, the storage doesn't move. It releases from original node but never makes it to the new/next node.

Like others, switching to `t2.xlarge` instances fixed this for me.

**Comment by gabordk** [comment]
Same problem here, AWS, k8s version 1.13.0

**Comment by brynmathias** [comment]
Same problem as well, AWS, eks. k8s version 1.11.0
See the issue on c5.9xlarge machines and on p2.xlarge instances.

I have a feeling that it might be due to the maximum number of ebs attachments to an ec2 node.
I do not know if the maximum attachment limit is total, or the attachments are properly removed.

**Comment by maflaven** [comment]
I've been seeing a similar issue on EKS with k8s version 1.11. Our support agent suggested the following:
```
It's very likely that the Kubernetes scheduler was choosing worker nodes in Availability Zones (AZ) with no volumes available. This can happen when the node selected for placement of the pod by the scheduler is not in the availability zone in which the Persistent volume(s) claim are available i.e. EBS volume exists. For example, when there isn't sufficient CPU and/or memory resources available on Nodes in which the persistent volume exists. That would lead to the scheduler choosing a Node in another zone and failing to schedule pod with this error. 

This was a known issue in Kubernetes[1][2] and has been fixed by having "VolumeScheduling"[2] feature enabled in scheduler.

Another workaround could include creating the volumes manually and updating the PVCs but both options would turn into a less available/dynamic cluster. 

References:
[1] https://github.com/kubernetes/enhancements/issues/490 
[2] https://github.com/kubernetes/kubernetes/issues/34583 
[3] https://kubernetes.io/blog/2018/10/11/topology-aware-volume-provisioning-in-kubernetes/ 
```

**Comment by gabordk** [comment]
Unfortunately this has nothing to do with AWS Availability Zones or VolumeScheduling. AZ related problems are kinda hot nowadays, so people like to mix up that problem with this one, but a quick look at the Availability Zones makes clear that there is no connection.

Today's testing results:
Kubernetes: 1.13.0
Instance: m5.4xlarge
EBS: gp2
- Both the worker node and the EBS volume is in the same AZ (us-east-1c) (VolumeScheduling is enabled in our StorageClass, previously it wasn't, nothing changed).
- EBS volume is successfully created and mounted on the worker node
- kubectl describe pv reports the volume as "Bound", events are empty
- kubectl describe pvc reports as "Bound", events are empty
- pod events:
        - MountVolume.WaitForAttach failed for volume "pvc-xxxxx" : could not find attached AWS Volume "aws://us-east-1c/vol-xxxxx". Timeout waiting for mount paths to be created
        - Unable to mount volumes for pod "xyz": timeout expired waiting for volumes to attach or mount for pod

**Comment by gabordk** [comment]
After some debugging I found my problem is this, all the symptoms are matching:
https://github.com/coreos/bugs/issues/2371

**Comment by hustshawn** [comment]
@gabordk I got the same issue. Any solution or progress on this?

**Comment by dfang** [comment]
same issue here.

on microk8s, enabled default storage addon,

then install helm-consul got this issue, all pv and pvc are bound. but consul-server pods still got "pod has unbound immediate PersistentVolumeClaims",  recreate these pods didn't help

**Comment by jhoblitt** [comment]
I believe I'm seeing the same issue on aws w/ eks 1.3.8 (1.3 "eks.2")

```
  Warning  FailedMount             56s (x11 over 23m)  kubelet, ip-192-168-124-20.ec2.internal  Unable to mount volumes for pod "jenkins-6758665c4c-gg5tl_jenkins(f6440463-ca87-11e9-a31c-0a4da4f89c32)": timeout expired waiting for volumes to attach or mount for pod "jenkins"/"jenkins-6758665c4c-gg5tl". list of unmounted volumes=[jenkins-home]. list of unattached volumes=[plugins tmp jenkins-config plugin-dir secrets-dir jenkins-home sc-config-volume jenkins-token-pn7mq]
```

**Comment by ishantanu** [comment]
@jhoblitt Did you tried with EKS 1.2.x? I faced some issues with statefulsets with PVC on EKS 1.3.x but everything ran just fine on EKS 1.2.x.

**Comment by mogaal** [comment]
@jhoblitt I was facing the same issue until 10 min ago, I realised there is a problem/bug with the kubernetes-plugin I was using. I solved upgrading to 1.18.3 the kubernetes-plugin for Jenkins.

**Comment by jhoblitt** [comment]
@ishantanu I don't believe I was seeing this problem with 1.2 but it's been awhile since I've tested with that version.

@mogaal This problem is present outside of pods managed by jenkins.
